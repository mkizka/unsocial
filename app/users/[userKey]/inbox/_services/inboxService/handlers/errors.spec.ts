import { z } from "zod";

import {
  ActivitySchemaValidationError,
  BadActivityRequestError,
  UnexpectedActivityRequestError,
} from "./errors";

describe("errors", () => {
  test(ActivitySchemaValidationError.name, () => {
    const parsed = z.object({ foo: z.string() }).safeParse({ foo: null });
    if (parsed.success) throw new Error();
    const error = new ActivitySchemaValidationError(parsed.error);
    expect(error.message).toMatchSnapshot();
    expect(error.statusCode).toBe(400);
  });
  test(BadActivityRequestError.name, () => {
    const error = new BadActivityRequestError("エラーメッセージ");
    expect(error.message).toBe("エラーメッセージ");
    expect(error.statusCode).toBe(400);
  });
  test(UnexpectedActivityRequestError.name, () => {
    const error = new UnexpectedActivityRequestError("エラーメッセージ");
    expect(error.message).toBe("エラーメッセージ");
    expect(error.statusCode).toBe(500);
  });
});
