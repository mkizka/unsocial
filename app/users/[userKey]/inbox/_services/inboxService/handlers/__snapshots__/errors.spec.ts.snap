// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`errors ActivitySchemaValidationError 1`] = `"Validation error: Expected string, received null at "foo""`;
